Migration guide for Keycloak admin client (#20091)
keycloak/keycloak: docs/documentation/upgrading/topics/keycloak/changes-22_0_0.adoc

> In previous releases the `export` and `import` commands allowed runtime parameters like, for example, a database URL only in configuration files or environment variables.

Reviewer: ```suggestion
In previous releases the `export` and `import` commands allowed runtime parameters such as a database URL only in configuration files or environment variables.
```

Starting with this release, those runtime parameters are now available on the command line as well.
Use the `--help` option to find out about the supported parameters.

> = Renamed Keycloak Admin client artifacts

Reviewer: ```suggestion
= Renamed Keycloak Admin Console artifacts
```

> = Renamed Keycloak Admin client dependencies

Reviewer 2: Actually, one more. Sorry, missed it before.
```suggestion
= Renamed Keycloak Admin client artifacts
```

Author: Oh, me too. Thanks @vmuzikar!

> After the upgrade to Jakarta EE, artifacts for Keycloak Admin clients were renamed to describe the reality more accurately, with consideration for long-term maintainability.
> We still provide two separate Keycloak Admin clients, one with Jakarta EE and the other with Java EE support.
> 

Reviewer: ```suggestion
After the upgrade to Jakarta EE, artifacts for Keycloak Admin Console were renamed to more descriptive names with consideration for long-term maintainability.
We still provide two separate Keycloak Admin Consoles, one with Jakarta EE and the other with Java EE support.

```

We stopped releasing the `org.keycloak:keycloak-admin-client-jakarta` artifact.

> The default one for Keycloak Admin client with Jakarta EE support is `org.keycloak:keycloak-admin-client` (since version 22.0.0).

Reviewer: ```suggestion
The default one for the Keycloak Admin Console with Jakarta EE support is `org.keycloak:keycloak-admin-client` (since version 22.0.0).
```

> We will also release Keycloak Admin client with Java EE support for some time, but it's recommended to migrate to Jakarta EE as soon as possible.

Reviewer: ```suggestion
We will continue to release the Keycloak Admin Console with Java EE support for some time, but we recommend that you migrate to Jakarta EE as soon as possible.
```

> = Renamed Keycloak Admin client dependencies
> 
> After the upgrade to JakartaEE, dependencies for Keycloak Admin clients were renamed to describe the reality more accurately, with consideration for long-term maintainability.
> We still provide two separate Keycloak Admin clients, one with JakartaEE and the other with JavaEE support.
> 
> We stopped releasing the `org.keycloak:keycloak-admin-client-jakarta` dependency.
> The default one for Keycloak Admin client with JakartaEE support is `org.keycloak:keycloak-admin-client` (since version 22.0.0).
> 
> We will also release Keycloak Admin client with JavaEE support for some time, but it's recommended to migrate to JakartaEE as soon as possible.
> The new dependency with JavaEE support is `org.keycloak:keycloak-admin-client-jee`.
> 
> === JakartaEE support
> [source,xml]
> .Before migration:
> ----
> <dependency>
>     <groupId>org.keycloak</groupId>
>     <artifactId>keycloak-admin-client-jakarta</artifactId>
>     <version>21.0.0</version>
> </dependency>
> ----
> 
> [source,xml]
> .After migration:
> ----
> <dependency>
>     <groupId>org.keycloak</groupId>
>     <artifactId>keycloak-admin-client</artifactId>
>     <version>22.0.0</version>
> </dependency>
> ----
> 
> === JavaEE support _(only temporary support)_
> [source,xml]
> .Before migration:
> ----
> <dependency>
>     <groupId>org.keycloak</groupId>
>     <artifactId>keycloak-admin-client</artifactId>
>     <version>21.0.0</version>
> </dependency>
> ----
> 
> [source,xml]
> .After migration:
> ----
> <dependency>
>     <groupId>org.keycloak</groupId>
>     <artifactId>keycloak-admin-client-jee</artifactId>
>     <version>22.0.0</version>
> </dependency>
> ----

Reviewer 2: JakartaEE --> Jakarta EE
JavaEE --> Java EE
```suggestion
= Renamed Keycloak Admin client dependencies

After the upgrade to Jakarta EE, dependencies for Keycloak Admin clients were renamed to describe the reality more accurately, with consideration for long-term maintainability.
We still provide two separate Keycloak Admin clients, one with Jakarta EE and the other with Java EE support.

We stopped releasing the `org.keycloak:keycloak-admin-client-jakarta` dependency.
The default one for Keycloak Admin client with Jakarta EE support is `org.keycloak:keycloak-admin-client` (since version 22.0.0).

We will also release Keycloak Admin client with Java EE support for some time, but it's recommended to migrate to Jakarta EE as soon as possible.
The new dependency with Java EE support is `org.keycloak:keycloak-admin-client-jee`.

=== Jakarta EE support
[source,xml]
.Before migration:
----
<dependency>
    <groupId>org.keycloak</groupId>
    <artifactId>keycloak-admin-client-jakarta</artifactId>
    <version>21.0.0</version>
</dependency>
----

[source,xml]
.After migration:
----
<dependency>
    <groupId>org.keycloak</groupId>
    <artifactId>keycloak-admin-client</artifactId>
    <version>22.0.0</version>
</dependency>
----

=== Java EE support _(only temporary support)_
[source,xml]
.Before migration:
----
<dependency>
    <groupId>org.keycloak</groupId>
    <artifactId>keycloak-admin-client</artifactId>
    <version>21.0.0</version>
</dependency>
----

[source,xml]
.After migration:
----
<dependency>
    <groupId>org.keycloak</groupId>
    <artifactId>keycloak-admin-client-jee</artifactId>
    <version>22.0.0</version>
</dependency>
----
```
